Plugins/GitUIPluginInterfaces/ObjectId.cs

```diff
@@ -326,6 +327,45 @@ public bool Equals(ObjectId? other)
         /// <inheritdoc />
         public override int GetHashCode() => unchecked((int)_i2);
 
+        public bool Equals(string? other)
+        {
+            if (other == null || other.Length != Sha1CharCount)
+            {
+                return false;
+            }
+
+            return PartToString(1).AsSpan().CompareTo(other.AsSpan(0, 16), StringComparison.InvariantCulture) == 0
```

Comment:
It'd be great to have comments explaining the algorithm here so that anyone who happens to stumble upon this code won't have as steep learning curve (and thus, has lesser risk of breaking things).